backend/de.metas.business/src/main/java/de/metas/tax/api/impl/TaxDAO.java

```diff
@@ -213,4 +190,15 @@ public ITranslatableString getTaxCategoryNameById(@Nullable final TaxCategoryId
 
 		return Optional.ofNullable(taxCategoryId);
 	}
+
+	@Override
+	public Optional<BigDecimal> getRateById(@NonNull final TaxId taxId)
```

Comment:
i think Optional is not needed because tax record shall always exist for a given ID. If not, it's a db inconsistency and not a missing rate.

But, instead of BigDecimal, pls consider returning de.metas.util.lang.Percent

Reply:
done